.github/workflows/robot-server-lint-test.yaml

```
@@ -76,8 +76,12 @@ jobs:
         run: make -C robot-server setup-ot2
       - name: Lint
         run: make -C robot-server lint
-      - name: Test
+      - if: ${{ matrix.with-ot-hardware == 'false' }}
+        name: Test without opentrons_hardware
         run: make -C robot-server test-cov test_opts="-m 'not ot3_only'"
+      - if: ${{ matrix.with-ot-hardware == 'true' }}
+        name: Test with opentrons_hardware
+        run: make -C robot-server test-cov test_opts="-m 'not ot2_only'"
```

Nice catch.

The original addition of `not ot2_only` feels weird to me. Why are we excluding entire server tests when `opentrons_hardware` is missing, anyway? If anything, shouldn't we test that the endpoint returns an explicit "Not an OT-3" error, or something?

Anyway. For this PR, when we're testing with `opentrons_hardware`, I think we should just:

```suggestion
      - if: ${{ matrix.with-ot-hardware == 'true' }}
        name: Test with opentrons_hardware
        run: make -C robot-server test-cov
```